Complex unittests

rubensayshi opened the pull request.

rubensayshi committed:
use an actual UTXO set for tests.
use normalized txIds to prevent fixtures updating based on order.
remove the `rawtransactions.db` file, instead use a memory DB.

rubensayshi committed:
add --dryrun and file match to copyscenarios tool

rubensayshi committed:
test arc4 and mocking enabled/disabled

rubensayshi committed:
updated fixtures

rubensayshi committed:
tool to update tx_ids in vectors

rubensayshi committed:
dont mock arc4 for unit_test

rubensayshi commented on .gitignore:
```diff
@@ -9,6 +9,7 @@ profile.txt
 
 # files generated by py.test
 /counterpartylib/test/fixtures/scenarios/*.new.*
+/counterpartylib/test/fixtures/rawtransactions.db
```
bye useless file

rubensayshi commented on counterpartylib/lib/api.py:
```diff
@@ -690,7 +690,7 @@ def search_raw_transactions(address, unconfirmed=True):
         @dispatcher.add_method
         def get_unspent_txouts(address, unconfirmed=False, unspent_tx_hash=None):
-            return backend.get_unspent_txouts(address, unconfirmed=unconfirmed, multisig_inputs=False, unspent_tx_hash=unspent_tx_hash)
+            return backend.get_unspent_txouts(address, unconfirmed=unconfirmed, unspent_tx_hash=unspent_tx_hash)
```
unused argument is unsed

rubensayshi commented on counterpartylib/lib/blocks.py:
```diff
@@ -492,7 +492,7 @@ def get_address(scriptpubkey):
             if ctx.is_coinbase():
                 raise DecodeError('coinbase transaction')
-            obj1 = ARC4.new(ctx.vin[0].prevout.hash[::-1])
+            obj1 = arc4.init_arc4(ctx.vin[0].prevout.hash[::-1])
```
moved into arc4 module so we can mock it

rubensayshi commented on counterpartylib/test/arc4_test.py:
```diff
@@ -0,0 +1,142 @@
+    assert k.decrypt(b'\xaa}\xfa5\xcaY:') == text
+
+
+@conftest.add_fn_property(DISABLE_ARC4_MOCKING=True)
```
this is how to disable the mocking

rubensayshi commented on counterpartylib/test/conftest.py:
```diff
@@ -69,16 +78,17 @@ def pytest_addoption(parser):
-    if (request.module.__name__ == 'integration_test'):
-        util_test.initialise_rawtransactions_db(db)
+    db = apsw.Connection(':memory:')
+    util_test.initialise_rawtransactions_db(db)
```
always just init in memory with fresh .sql load

rubensayshi commented on counterpartylib/test/conftest.py:
```diff
@@ -126,20 +138,139 @@ def cp_server(request):
+        util_test.MOCK_UTXO_SET = MockUTXOSet(json.load(listunspent_test_file), rawtransactions_db)
+
+    util_test.UNIQUE_DUMMY_TX_HASH = {}
+    util_test.TX_INDEX = 1
```
`scope=function, autouse=True` so this is where we reset stuff between each test

rubensayshi commented on counterpartylib/test/util_test.py:
```diff
@@ -132,10 +139,15 @@ def insert_block(db, block_index, parse_block=True):
     if parse_block:
         blocks.parse_block(db, block_index, block_time)
+
+    MOCK_UTXO_SET.increment_confirmations()
```
bumps the utxos in the mock set

rubensayshi commented on counterpartylib/test/util_test.py:
```diff
@@ -146,35 +158,100 @@ def create_next_block(db, block_index=None, parse_block=True):
+
+
+UNIQUE_DUMMY_TX_HASH = {}
+def dummy_tx_hash(raw_transaction):
```
create a blank version of the tx to normalize the txids as much as possible

rubensayshi committed:
add mocked `backend.`getrawtransaction_batch` and `backend.searchrawtransactions`.

rubensayshi committed:
make ConfigContext support non existant properties too

rubensayshi committed:
adjust complex unit test for fee/kb

rubensayshi committed:
avoid UTXO lock being set and then bytespersigop fix kicking in

rubensayshi committed:
atempt to get around circle CI timeout

unsystemizer committed:
Add notes on 'force' and 'sudo'

- `--force` added to the server configuration section
- Note on sudo access added to the install section
- Addresses https://github.com/CounterpartyXCP/Documentation/issues/39

unsystemizer committed:
Add that --force is for non-production servers

littleskunk committed:
allow estimate_fee_per_kb and estimate_fee_per_kb_nblocks as arguments

littleskunk committed:
estimate fee_per_kb for nblock > 10

deweller committed:
Implements Subassets cip-4

https://github.com/CounterpartyXCP/cips/blob/master/cip-0004.md

deweller committed:
Move new column bindings to the end of the list.

The order of the additional field and bindings matter when adding a new column.

deweller committed:
Move asset_longname column in tests to match code

deweller committed:
Always create index if it does not exist.

Keep consistency with the other CREATE INDEX lines.

deweller committed:
variable name consistency improvements

deweller committed:
Fix assets table creation

deweller committed:
subasset_longname variable consistency.

deweller committed:
Prevent crash where status is invalid due to nonexistent reissuance variable

deweller committed:
coding standards

deweller committed:
Regenerate test fixtures with new assets table creation

deweller committed:
Handle reissuance composition and parsing

deweller committed:
Changelog

deweller committed:
Implements subasset issuances as a separate transaction type

deweller committed:
Improved logging of subasset issuance

deweller committed:
Force big-endian decoding of subasset name and description

deweller committed:
Add explicit check of subasset_longname validation while parsing issuances

deweller committed:
Support subassets in more API methods (#1)

support subassets in the create_send, create_order, create_dividend and create_destroy methods.

deweller committed:
Accept subasset longname in a few more API methods

robby-d committed:
API: don't override fee_per_kb if specified. also better handling of invalid parameters passed to a create method

deweller committed:
A little documentation with the subasset longname encoding

deweller committed:
Don’t parse issuance issuance messages before activation

deweller committed:
raise UnpackError for SUBASSET_ID messages when subassets are not available

robby-d committed:
update bitcoin lib deps for segwit support

robby-d committed:
use python-bitcoinlib 0.7.1-SNAPSHOT for segwit support

deweller committed:
Change block indexes and rebuild all scenario message hashes

Commit:
Merge pull request #961 from deweller/cip4-wip-2

Implements Subassets cip-4

F483 commented:
Hi, any reason why this hasn't been merged yet? The micropayments patch depends on this patch.

robby-d committed:
Revert "use python-bitcoinlib 0.7.1-SNAPSHOT for segwit support"
Move this into its own branch until PTodd reviews and pushes the code

This reverts commit c37e01829d633fb8135c847f599bd8754d311621.

robby-d committed:
update reparsing to be less verbose, and perform a VACUUM afterwards for better subsequent performance

robby-d committed:
add note on bitcoind update to changelog

robby-d committed:
set protocol change block for CIP4 activation

robby-d committed:
expose database.vacuum in server.py

robby-d committed:
update date of new version

robby-d committed:
we have to use python-bitcoinlib 0.7.1-SNAPSHOT on testnet, but ONLY use it on testnet
(this is a hack works with the docker containers -- manual installations will need to manually switch out the packages)

robby-d committed:
tweak python-bitcoinlib version specifier

robby-d committed:
update python-bitcoinlib version range

robby-d committed:
correct download URL

robby-d committed:
take a different approach with python-bitcoinlib

robby-d committed:
move python-bitcoinlib repo to CounterpartyXCP

robby-d committed:
require setuptools-markdown

robby-d committed:
hotfix for testnet reparses raising a ConsensusError on retroactive protocol changes

robby-d committed:
fixes #969

robby-d committed:
further fixes #969

robby-d committed:
modified UTXO selection algo (#969)

robby-d committed:
further tweak UTXO selection

robby-d committed:
more UTXO selection tweaks (this algo is more like the old version, except we are filtering out multisig dust too)

rubensayshi committed:
upgrade pip before installing requirements

cryptcoin-junkey committed:
Rename docker-engine to docker-ce. (#982)

deweller committed:
Fix testnet consensus hash for missing testnet transaction

Transaction 8aa75e44c718a22b2c4fc6abad4e9a0b2721032f554120df52e8632aa10bcdf5 in block 626250 was not included in the previous hashes.

deweller committed:
upgrade pip before installing requirements

From @rubensayshi commit 6c6dc776b4611224799afa338b33c37534006072
